.github/workflows/test.yml

```diff
@@ -24,13 +24,13 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
       - name: Python setup
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python_version }}
       # remove poetry.lock file for scheduled tests
       # to help simulate possible upstream issues
       - name: Remove poetry.lock for scheduled tests
-        if: github.event_name == 'schedule'
+        if: github.event.schedule == '0 7 * * 3'
```

Comment:
It might be worth adding a bit of detail in a comment specifying what this means exactly

Reply:
Thanks for this comment! I made this change as initially I felt that the test might be so large that it'd warrant an additional schedule (in addition to the Wednesday scheduled runs). This was a way to specify exactly which schedule was used as a trigger for a task within the job when there might be multiple. I retained it because it felt more declarative and we could end up with more scheduled tests yet in the future. I'll add a comment as you suggested to keep this well documented (`crontab` on its own is a bit terse to read!).